### PR TITLE
Defensively handle DocumentAPI calls

### PR DESCRIPTION
**Pull request by neilmendum**
https://hackney.atlassian.net/browse/DES-202

### CODE REVIEW

File: EvidenceApi.Tests/V1/E2ETests/DocumentSubmissionsTest.cs [hunk 1]
```diff
@@ -132,6 +132,87 @@ public async Task UnsuccessfulWithInvalidParams()
             response.StatusCode.Should().Be(400);
         }
 
+        [Test]
+        public async Task CreateDocumentSubmissionUnsuccessfulWhenDocumentApiCannotCreateClaim()
```
Comment (bogdan-zaharia) [inline EvidenceApi.Tests/V1/E2ETests/DocumentSubmissionsTest.cs:136]: I think test names should be general and not depend on the underlying functionality/system. In you still want to go ahead with this name:
```suggestion
        public async Task CreateDocumentSubmissionUnsuccessfulWhenDocumentsApiCannotCreateClaim()
```

File: EvidenceApi.Tests/V1/E2ETests/DocumentSubmissionsTest.cs [hunk 2]
```diff
@@ -132,6 +132,87 @@ public async Task UnsuccessfulWithInvalidParams()
...
+                .Without(x => x.DocumentSubmissions)
+                .Create();
+            DatabaseContext.EvidenceRequests.Add(entity);
+            DatabaseContext.SaveChanges();
+
+            DocumentsApiServer.Reset();
+            DocumentsApiServer.Given(
+                Request.Create().WithPath("/api/v1/claims")
+            ).RespondWith(
+                Response.Create().WithStatusCode(404)
```
Comment (bogdan-zaharia) [inline EvidenceApi.Tests/V1/E2ETests/DocumentSubmissionsTest.cs:152]: Nice one!

File: EvidenceApi.Tests/V1/E2ETests/DocumentSubmissionsTest.cs [hunk 3]
```diff
@@ -226,6 +307,32 @@ public async Task ReturnNotFoundWhenCannotFindDocumentSubmission()
             response.StatusCode.Should().Be(404);
         }
 
+        [Test]
+        public async Task ReturnBadRequestWhenDocumentsApiHasIssueDuringFindDocumentSubmission()
```
Comment (bogdan-zaharia) [inline EvidenceApi.Tests/V1/E2ETests/DocumentSubmissionsTest.cs:311]: Same comment as above about the name

File: EvidenceApi.Tests/V1/UseCase/CreateDocumentSubmissionUseCaseTests.cs [hunk 4]
```diff
@@ -54,6 +54,70 @@ public void ThrowsNotFoundExceptionWhenEvidenceRequestIsNull()
             testDelegate.Should().Throw<NotFoundException>();
         }
 
+        [Test]
+        public void ThrowsBadRequestExceptionWhenDocumentApiCannotCreateClaim()
```
Comment (bogdan-zaharia) [inline EvidenceApi.Tests/V1/UseCase/CreateDocumentSubmissionUseCaseTests.cs:58]: same as above

File: EvidenceApi.Tests/V1/UseCase/CreateDocumentSubmissionUseCaseTests.cs [hunk 5]
```diff
@@ -54,6 +54,70 @@ public void ThrowsNotFoundExceptionWhenEvidenceRequestIsNull()
...
+
+            // Act
+            Func<Task<DocumentSubmissionResponse>> testDelegate = async () => await _classUnderTest.ExecuteAsync(evidenceRequest.Id, _request).ConfigureAwait(true);
+
+            // Assert
+            testDelegate.Should().Throw<BadRequestException>();
+        }
+
+        [Test]
+        public void ThrowsBadRequestThrowsNotFoundExceptionWhenDocumentApiCannotCreateUploadPolicy()
```
Comment (bogdan-zaharia) [inline EvidenceApi.Tests/V1/UseCase/CreateDocumentSubmissionUseCaseTests.cs:86]: same as above or:
```suggestion
        public void ThrowsBadRequestThrowsNotFoundExceptionWhenDocumentsApiCannotCreateUploadPolicy()
```

File: EvidenceApi.Tests/V1/UseCase/FindDocumentSubmissionByIdUseCaseTests.cs [hunk 6]
```diff
@@ -81,6 +81,20 @@ public void ThrowsAnErrorWhenADocumentSubmissionIsNotFound()
             testDelegate.Should().Throw<NotFoundException>().WithMessage($"Cannot find document submission with ID: {id}");
         }
 
+        [Test]
+        public void ThrowsAnErrorWhenDocumentApiCannotFindClaim()
```
Comment (bogdan-zaharia) [inline EvidenceApi.Tests/V1/UseCase/FindDocumentSubmissionByIdUseCaseTests.cs:85]: same as above or:
```suggestion
        public void ThrowsAnErrorWhenDocumentsApiCannotFindClaim()
```

File: EvidenceApi.Tests/V1/E2ETests/DocumentSubmissionsTest.cs (hunk 3 above)
Comment (neilmendum) [reply EvidenceApi.Tests/V1/E2ETests/DocumentSubmissionsTest.cs]: Good point! addressed

File: EvidenceApi.Tests/V1/UseCase/CreateDocumentSubmissionUseCaseTests.cs (hunk 4 above)
Comment (neilmendum) [reply EvidenceApi.Tests/V1/UseCase/CreateDocumentSubmissionUseCaseTests.cs]: Good point! addressed

File: EvidenceApi.Tests/V1/UseCase/CreateDocumentSubmissionUseCaseTests.cs (hunk 5 above)
Comment (neilmendum) [reply EvidenceApi.Tests/V1/UseCase/CreateDocumentSubmissionUseCaseTests.cs]: Good point! addressed

File: EvidenceApi.Tests/V1/UseCase/FindDocumentSubmissionByIdUseCaseTests.cs (hunk 6 above)
Comment (neilmendum) [reply EvidenceApi.Tests/V1/UseCase/FindDocumentSubmissionByIdUseCaseTests.cs]: Good point! addressed

**Comment by neilmendum** [comment]
Cheers dude, have addressed those

**Review by bogdan-zaharia — APPROVED** [review]
👍

**Pull request merged**